contrib/infiniband/ibvctx.c

```diff
@@ -1104,9 +1106,34 @@ int _get_cq_event(struct ibv_comp_channel * channel,
 {
   struct internal_ibv_comp_channel * internal_channel;
   struct internal_ibv_cq * internal_cq;
-  int rslt;
+  int rslt, flags;
 
   internal_channel = ibv_comp_to_internal(channel);
+  flags = fcntl(internal_channel->real_channel->fd, F_GETFL, NULL);
+
+  // We need to change the call to non-blocking mode
+  if (flags & O_NONBLOCK == 0) {
+    int ms_timeout = 100;
+    struct pollfd my_pollfd = {
+      .fd = internal_channel->real_channel->fd,
+      .events = POLLIN,
+      .revents = 0
```

Comment:
For consistency with the rest of DMTCP, could we use the older style:
```
struct pollfd my_pollfd;
my_pollfd.fd = internal_channel->real_channel->fd;
```
and so on.  (If others have opinions about starting to use the new syntax now, feel free to add your comments.)